### PR TITLE
feat: merge upstream template v2.4.0

### PR DESCRIPTION
**Pull request by jens-getpassport**
Based on https://github.com/getpassport/template_services_python/pull/138 we fixed and changed a few minor issues in our upstream template, and now merge them into this template repo as well.

### CODE REVIEW

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -17,8 +17,7 @@ updates:
   # Add additional reviewers for PRs opened by Dependabot. For more information, see:
   # https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#reviewers
   reviewers:
-  - brad-getpassport
-  - jens-getpassport
+  - template-maintainers
```
Comment (brad-getpassport) [inline .github/dependabot.yaml:20]: This requires that this group be created in `github` so hopefully the README has been modified.

File: .github/workflows/pull-request.yaml [hunk 2]
```diff
@@ -8,7 +8,6 @@ on:
   pull_request:
     branches:
     - main
-    - staging
```
Comment (brad-getpassport) [inline .github/workflows/pull-request.yaml:11]: Why did we remove the staging branch?  We probably want a UAT branch and I was thinking that was staging... or are you thinking we use `develop` as the staging?

File: README.md [hunk 3]
```diff
@@ -59,6 +59,8 @@ Automatic package versioning and tagging, publishing to [PyPI](https://pypi.org/
 
 [OSSF Security Scorecards](https://github.com/ossf/scorecard) is enabled as a GitHub Actions workflow to give the consumers information about the supply-chain security posture of this project, assigning a score of 0–10. We upload the results as a SARIF (Static Analysis Results Interchange Format) artifact after each run and the results can be found at the Security tab of this GitHub project. We also allow publishing the data at [OpenSSF](https://metrics.openssf.org/). We use this data to continuously improve the security posture of this project. Note that this configuration supports the ``main`` (default) branch and requires the repository to be public and not forked.
 
+[pip-audit](https://github.com/pypa/pip-audit) is part of the default Python virtual environment, and can be used to check all installed packages for documented [CVE](https://www.cve.org/) by querying the [Python Packaging Advisory Database](https://github.com/pypa/advisory-database). The `build.yaml` workflow always runs a package audit before the artifacts are being built.
+
```
Comment (brad-getpassport) [inline README.md:63]: Probably need to update the README to call out that they should create a `template-maintainer` group in Github or change the line?

File: .github/workflows/pull-request.yaml (hunk 2 above)
Comment (jens-getpassport) [reply .github/workflows/pull-request.yaml:11]: > Why did we remove the staging branch?

Our current branching model with Passport doesn’t contain a `staging` branch, and I didn’t want to carry over a branch name from a different repository that doesn’t exist.

> We probably want a UAT branch and I was thinking that was staging...

The current `develop` branch is for UAT, isn’t it?

> or are you thinking we use develop as the staging?

I’m unsure, actually. When I use `staging` in the other repo then it’s to accumulate and merge PRs before we “flush” them into `main` for a new release. At the moment we use `develop` to do the same… 🤔

File: README.md (hunk 3 above)
Comment (jens-getpassport) [reply README.md:63]: This comment is related to [the other one](https://github.com/getpassport/template_package_python/pull/38#discussion_r967681171)?

File: .github/dependabot.yaml (hunk 1 above)
Comment (jens-getpassport) [reply .github/dependabot.yaml:20]: Building on the changes from PR https://github.com/getpassport/template_package_python/pull/37, right?

Comment (jens-getpassport) [reply .github/dependabot.yaml:20]: Commit https://github.com/getpassport/template_package_python/pull/38/commits/07f52ff5880c8a9bf6a066dd1820c8cae941c932. I wonder if we should use `package-maintainer` in the actual package repos?

**Review by brad-getpassport — APPROVED** [review]

**Pull request merged**